src/olympia/devhub/tasks.py

```diff
@@ -281,6 +292,21 @@ def annotate_upgrade_warning(results, file_, addon, version_string):
             'message': msg,
             'description': [],
             'compatibility_type': None})
+        results['warnings'] += 1
```

Comment:
Yes, it's guaranteed that it's defined. `devhub.utils:fix_addons_linter_output` set's it explicitly and relies on it too